src/rgw/setup/rgw.py

```diff
@@ -536,40 +536,41 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
-            user_status = Rgw._create_admin_on_current_node(conf, current_data_node)
-
-            if user_status == 0:
-                Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
-                Rgw._set_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY, const.ADMIN_USER_CREATED)
-            else:
-                machine_ids = Rgw._get_cortx_conf(conf, const.MACHINE_IDS_KEY)
-                data_pod_hostnames = [Rgw._get_cortx_conf(conf,
-                    f'node>{machine_id}>hostname') for machine_id in machine_ids if
-                    Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
-                if len(data_pod_hostnames) == 1 and current_data_node == data_pod_hostnames[0]:
-                    Log.error('Admin user creation failed')
-                    Rgw._delete_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY)
-                    raise SetupError(user_status, 'Admin user creation failed on'
-                        f' "{Rgw._machine_id}" node, with all data pods - {data_pod_hostnames}')
-
-                data_pod_hostnames.remove(current_data_node)
-                for data_pod_hostname in data_pod_hostnames:
-                    try:
-                        Rgw._update_hax_endpoint(conf, data_pod_hostname)
-                    except Exception:
-                        continue
-                    status = Rgw._create_rgw_user(conf)
-                    if status == 0:
-                        Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
-                        Rgw._set_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY, const.ADMIN_USER_CREATED)
-                        break
-                    else:
-                        if data_pod_hostname == data_pod_hostnames[-1]:
-                            Log.error(f'Admin user creation failed with error code - {status}')
-                            Rgw._delete_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY)
-                            raise SetupError(status, 'Admin user creation failed on'
-                                f' "{Rgw._machine_id}" node, with all data pods - {data_pod_hostnames}')
+            # TODO: Find a way to get current data pod hostname on server node.
+            # current_data_node = socket.gethostname().replace('server', 'data')
+            # user_status = Rgw._create_admin_on_current_node(conf, current_data_node)
+
+            # if user_status == 0:
+            #    Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
+            #    Rgw._set_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY, const.ADMIN_USER_CREATED)
+            # else:
+            machine_ids = Rgw._get_cortx_conf(conf, const.MACHINE_IDS_KEY)
+            data_pod_hostnames = [Rgw._get_cortx_conf(conf,
+                f'node>{machine_id}>hostname') for machine_id in machine_ids if
+            Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
```

Comment:
pls add this 'type' string to const.py.
there are discussions happening for changing the pod names.. tomorrow it might be some other value..so pls make it configurable in const.py

Reply:
Done